### DISCUSSION TITLE
How to access GetComm ?

### DISCUSSION THREAD

**Discussion by ampdes** [ornladios/ADIOS2#4296] [Q&A]
See the MWE on how I am trying to fetch the MPI_Comm associated with the adios.
I see an error GetComm not a member of adios2::ADIOS

What is the correct way?
Thanks.

```C++
#include <adios2.h>
#include <adios2/core/ADIOS.h>
#include <adios2/helper/adiosComm.h>
#include <adios2/helper/adiosCommMPI.h>
#include <mpi.h>

int main(int argc, char* argv[])
{

MPI_Init(&argc, &argv);

adios2::ADIOS adios(MPI_COMM_WORLD);

adios2::helper::Comm adios_comm = adios.GetComm();
MPI_Comm _comm = adios2::helper::CommAsMPI(adios_comm);

MPI_Finalize();
return 0;
}
```

**Comment by eisenhauer** [comment]
Can I ask about your desired usage here?  You are mixing APIs.  Generally we have a published/documented API produced by ADIOS' external bindings, but there also exists an internal API which isn't externally documented, may change between minor releases, etc.  This internal API is only installed because ADIOS supports the external development of engines and transports which might be dlopen'd and installed at run-time. ("Supports" might be a strong word, since this is an undocumented feature that might be considered experimental.)  The 'core' and 'helper' include files and namespaces are part of these internal APIs.  Externally, we don't expose an API to retrieve the MPI_Comm because it is received from the application, not generated by us.  If it weren't in general use by the application, the app probably wouldn't have initialized ADIOS with MPI.

**Reply by ampdes** [reply]
Desired usage is to read from a file and then perform some computations. One needs a comm for these computations as well.
Fetching the MPI_Comm from the ADIOS2 (initialized in the application with an MPI) would have been nice to have in the public API.

I guess then one just needs to pass around comm as well separately.

**Reply by eisenhauer** [reply]
Yes, since it was provided to us by the application, we kind of assume the app doesn't need an API to get it back because it can just remember what it gave us.  You can check out the examples directory in the source if that's helpful.
